packages/webassemblyjs/src/interpreter/runtime/values/module.js

```diff
@@ -292,39 +292,13 @@ function instantiateExports(
   traverse(n, {
     ModuleExport({ node }: NodePath<ModuleExport>) {
       if (node.descr.type === "Func") {
-        // Referenced by its index in the module.funcaddrs
-        if (node.descr.id.type === "NumberLiteral") {
```

Comment:
Could we maybe throw here? like "internal compiler failure". Because if we get into this it will be very hard to debug.

Reply:
Yeah, good idea - I've done a slightly more aggressive refactor in https://github.com/xtuc/webassemblyjs/pull/225/commits/942797a8e62e31d594b3dd85d1664d3d789fe9f9 so that all the temporary 'instantiated' items have the same structure. This means I was able add this exception logic in a single place.